src/network/networkd-address.c

```diff
@@ -787,7 +787,7 @@ int config_parse_address(const char *unit,
         else
                 n->in_addr_peer = buffer;
 
-        if (n->family == AF_INET && n->broadcast.s_addr == 0)
+        if (n->family == AF_INET && n->broadcast.s_addr == 0 && n->prefixlen <= 30)
                 n->broadcast.s_addr = n->in_addr.in.s_addr | htonl(0xfffffffflu >> n->prefixlen);
```

Comment:
This seems to be off-by-one:
```pycon
>>> 0xffffffff >> 31
1
```
Maybe write this as `... && n->prefixlen < 32`.

Reply:
When `prefixlen == 31` there are no broadcast address. So, it is not necessary to set that here.

Reply:
Hmm, when prefixlen==31, then we have two addresses, no? `b...0` and `b...1`. I know this is not very useful, but I think we should allow this for consistency.

Reply:
It is used for point-to-point. https://tools.ietf.org/html/rfc3021.

Reply:
OK. Thanks for the explanation.